{"metadata":{"role":"symbol","modules":[{"name":"TrustlessSDK"}],"externalID":"s:12TrustlessSDK11ResultValuea","title":"ResultValue","navigatorTitle":[{"text":"ResultValue","kind":"identifier"}],"fragments":[{"text":"typealias","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"ResultValue"}],"symbolKind":"typealias","roleHeading":"Type Alias"},"sections":[],"schemaVersion":{"patch":0,"major":0,"minor":3},"primaryContentSections":[{"declarations":[{"platforms":["iOS"],"tokens":[{"kind":"keyword","text":"typealias"},{"kind":"text","text":" "},{"text":"ResultValue","kind":"identifier"},{"kind":"text","text":"<"},{"kind":"genericParameter","text":"Value"},{"text":"> = ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:s6ResultO","text":"Result"},{"text":"<","kind":"text"},{"text":"Value","kind":"typeIdentifier","preciseIdentifier":"s:12TrustlessSDK11ResultValuea0D0xmfp"},{"text":", ","kind":"text"},{"text":"TrustlessError","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessError","kind":"typeIdentifier","preciseIdentifier":"s:12TrustlessSDK0A5ErrorO"},{"kind":"text","text":">"}],"languages":["swift"]}],"kind":"declarations"},{"kind":"parameters","parameters":[{"content":[{"inlineContent":[{"type":"text","text":"The type of the value in the success case."}],"type":"paragraph"}],"name":"Value"}]},{"content":[{"type":"heading","text":"Discussion","anchor":"discussion","level":2},{"type":"paragraph","inlineContent":[{"type":"text","text":"This type simplifies the handling of operations that can either succeed with a value of type "},{"code":"Value","type":"codeVoice"},{"type":"text","text":" or fail with a "},{"type":"codeVoice","code":"TrustlessError"},{"text":".","type":"text"},{"type":"text","text":" "},{"text":"It is used extensively across the SDK in functions that perform asynchronous tasks, such as network requests, which can either produce a result or an error.","type":"text"}]}],"kind":"content"}],"hierarchy":{"paths":[["doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK"]]},"abstract":[{"type":"text","text":"A typealias for the "},{"type":"codeVoice","code":"Result"},{"type":"text","text":" type specialized for the TrustlessSDK, where failures are represented by "},{"type":"codeVoice","code":"TrustlessError"},{"type":"text","text":"."}],"kind":"symbol","identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultValue"},"seeAlsoSections":[{"identifiers":["doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultCallback","doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/CancelCallback"],"generated":true,"title":"Request and Response"}],"variants":[{"paths":["\/documentation\/trustlesssdk\/resultvalue"],"traits":[{"interfaceLanguage":"swift"}]}],"references":{"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/ResultValue":{"url":"\/documentation\/trustlesssdk\/resultvalue","fragments":[{"text":"typealias","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"ResultValue"}],"title":"ResultValue","navigatorTitle":[{"text":"ResultValue","kind":"identifier"}],"abstract":[{"type":"text","text":"A typealias for the "},{"code":"Result","type":"codeVoice"},{"type":"text","text":" type specialized for the TrustlessSDK, where failures are represented by "},{"code":"TrustlessError","type":"codeVoice"},{"type":"text","text":"."}],"kind":"symbol","type":"topic","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultValue","role":"symbol"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessError":{"title":"TrustlessError","type":"topic","navigatorTitle":[{"text":"TrustlessError","kind":"identifier"}],"role":"symbol","url":"\/documentation\/trustlesssdk\/trustlesserror","abstract":[{"type":"text","text":"Describes errors that can occur within the TrustlessSDK."}],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"text":"TrustlessError","kind":"identifier"}],"kind":"symbol"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/CancelCallback":{"role":"symbol","abstract":[{"text":"A typealias for a closure used to cancel an ongoing operation.","type":"text"}],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/CancelCallback","url":"\/documentation\/trustlesssdk\/cancelcallback","type":"topic","navigatorTitle":[{"kind":"identifier","text":"CancelCallback"}],"kind":"symbol","title":"CancelCallback","fragments":[{"text":"typealias","kind":"keyword"},{"kind":"text","text":" "},{"text":"CancelCallback","kind":"identifier"}]},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK":{"title":"TrustlessSDK","url":"\/documentation\/trustlesssdk","role":"collection","type":"topic","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK","abstract":[],"kind":"symbol"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/ResultCallback":{"role":"symbol","abstract":[{"text":"A typealias for a callback function that is invoked with a ","type":"text"},{"code":"ResultValue","type":"codeVoice"},{"text":".","type":"text"}],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultCallback","url":"\/documentation\/trustlesssdk\/resultcallback","type":"topic","navigatorTitle":[{"text":"ResultCallback","kind":"identifier"}],"kind":"symbol","title":"ResultCallback","fragments":[{"text":"typealias","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"ResultCallback"}]}}}